lib/util.js

```diff
@@ -97,15 +97,29 @@ function applyDefault(defaultOptions, userOptions) {
     });
 
     return options;
-}
-exports.applyDefault = applyDefault;
+};
 
 /**
  * Upper cases the first character or the string
  * @param {string} str a string
  * @returns {string} upper case first
  */
-function upperCaseFirst(str) {
-    return str[0].toUpperCase() + str.slice(1);
-}
-exports.upperCaseFirst = upperCaseFirst;
+exports.upperCaseFirst = str => str[0].toUpperCase() + str.slice(1);
+
+/**
+ * Try to retrieve typescript parser service from context
+ * @param {RuleContext} context Rule context
+ * @returns {{esTreeNodeToTSNodeMap}|{program}|Object|*} parserServices
+ */
+exports.getParserServices = context => {
+    if (
+        !context.parserServices ||
+        !context.parserServices.program ||
+        !context.parserServices.esTreeNodeToTSNodeMap
+    ) {
+        throw new Error(
+            "This rule requires you to use `typescript-eslint-parser`."
```

Comment:
```suggestion
            "This rule requires you to use `eslint-plugin-typescript/parser`."
```

Reply:
ithink this message is wrong anyway, it requires project field in eslint parser config.

Reply:
does the user have to provide the project field?
it seems like a fiddly bit of config if it can just use the app root...

Reply:
hmm we'll have to figure it out once they have finalised everything.

I hope we can make that an optional thing, and have it use the root folder by default.
If not we will have to write up some decent docs on how to configure it.

Reply:
@bradzacher I just noticed this comment. Is there a different way you'd prefer the semantic rules to be configured? My original intent was to match tslint, where a tsconfig has to be provided in order to power the rules. But if there's a more implicit way that makes more sense, I'd be happy to try to work it in.

Reply:
Ideally I'd like to have it implicitly using the root tsconfig unless otherwise specified. This would make everything just work out of the box. 

Unfortunately right now if you want to use the plugin, you can; but if you want to use an upcoming rule that requires type info, you have to point it at your tsconfig.

This creates a confusing dichotomy in the configuration where it can both work and not work at the same time.

Additionally in my experience, explicitly defining paths in config files can sometimes be confusing depending on project folder structure, and what folder the tool considers the root.

So if we can make it work for the default case, then it'll be easy for users (which means less question type issues being opened).

Reply:
Got it. I should mention that the concern here is that the information we're providing through parser services incurs a major performance hit (as in TSLint). So I tried to make it as opt-in as possible so as to not completely sink users' lint times.
I'd be happy to help put together docs to make it as clear as we can, but I would like to maintain the opt-in nature of semantic rules.